crates/rustc_codegen_spirv/src/linker/capability_computation.rs

```diff
@@ -104,28 +72,36 @@ fn operand_required_extensions(op: &Operand) -> &'static [&'static str] {
 // It says no. We strip it. Things explode.
 // So, this function is to encode any special version-specific rules that aren't in rspirv.
 fn additional_extensions(module: &Module, inst: &Instruction) -> &'static [&'static str] {
-    if module.header.as_ref().unwrap().version() < (1, 5)
-        && inst.class.opcode == Op::Capability
-        && inst.operands[0].unwrap_capability() == Capability::VulkanMemoryModel
-    {
-        &["SPV_KHR_vulkan_memory_model"]
+    let version = module.header.as_ref().unwrap().version();
+    if inst.class.opcode == Op::Capability {
+        match inst.operands[0].unwrap_capability() {
+            Capability::VulkanMemoryModel if version < (1, 5) => &["SPV_KHR_vulkan_memory_model"],
+            Capability::RuntimeDescriptorArray => &["SPV_EXT_descriptor_indexing"],
+            _ => &[],
+        }
     } else {
         &[]
     }
 }
 
 pub fn remove_extra_extensions(module: &mut Module) {
-    let set: HashSet<&str> = module
+    let mut set: HashSet<&str> = module
         .all_inst_iter()
         .flat_map(|inst| {
             inst.class
                 .extensions
                 .iter()
-                .chain(inst.operands.iter().flat_map(operand_required_extensions))
                 .chain(additional_extensions(module, inst))
         })
         .copied()
         .collect();
+
+    set.extend(
+        module
+            .all_inst_iter()
+            .flat_map(|inst| inst.operands.iter().flat_map(|op| op.required_extensions())),
+    );
```

Comment:
Why did you split this out to its own thing? It's less efficient this way

Reply:
Iterator sadness - `required_extensions` became something that returns a Vec<> instead of a &'static slice now and I could get the chain calls to match up anymore.